fixtures/asset-config/auto-trailing-slash.test.ts

```diff
@@ -0,0 +1,226 @@
+import { SELF } from "cloudflare:test";
+import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
+import Worker from "../../packages/workers-shared/asset-worker/src/index";
+import { getAssetWithMetadataFromKV } from "../../packages/workers-shared/asset-worker/src/utils/kv";
+import type { AssetMetadata } from "../../packages/workers-shared/asset-worker/src/utils/kv";
+
+const IncomingRequest = Request<unknown, IncomingRequestCfProperties>;
+
+vi.mock("../../packages/workers-shared/asset-worker/src/utils/kv.ts");
+const existsMock = (fileList: Set<string>) => {
+	vi.spyOn(Worker.prototype, "exists").mockImplementation(
+		async (pathname: string) => {
+			if (fileList.has(pathname)) {
+				return pathname;
+			}
+		}
+	);
+};
+const BASE_URL = "http://example.com";
+
+type TestCase = {
+	title: string;
+	files: string[];
+	requestPath: string;
+	matchedFile?: string;
+	finalPath?: string;
+	// redirected?: boolean;
+};
+
+const testCases: TestCase[] = [
```

Comment:
These don't seem to capture the status codes they're testing against?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Request/redirect